source/manage/cloud-data-export.rst

```diff
@@ -1,8 +1,138 @@
+
+Authenticate
+------------
+
+Authentication is done with either Mattermost login credentials or an authentication token. First, use your administrator credentials to log into the instance with mmctl, replacing `example-source-domain.com` with the network address of the source instance:
+
+.. code::
+
+   $ mmctl auth login https://yourdomain.cloud.mattermost.com
+   
+You'll be prompted for a username (use your admin user), password, and for a connection name. The connection name can be anything you want, and it's used to identify this set of credentials in the future, for your convenience. Then you will be able to start the export process.
+
+Create the export
+-----------------
+
+Once you're logged in, run:
+
+.. code::
+
+   $ mmctl export create --attachments
+
+This will create a full export of the server, and include attached files. Leave out ``--attachments`` if you do not wish to export attached files from your instance. This process can take some time, so ``mmctl`` will return immediately, and the job will run in the background on the Mattermost instance until the export is fully created. If successful, the command will immediately output a job ID, like this:
+
+.. code::
+
+   Export process job successfully created, ID: yfrr9ku5i7fjubeshs1ksrknzc
+
+While the job is running, its status can be checked using the ID that was provided when it was created, and when it's done the output will look similar to this:
+
+.. code::
+
+  $ mmctl export job show yfrr9ku5i7fjubeshs1ksrknzc
+    ID: yfrr9ku5i7fjubeshs1ksrknzc
+    Status: success
+    Created: 2021-11-03 10:44:13 -0500 CDT
+    Started: 2021-11-03 10:44:23 -0500 CDT
+
+Download the export
+-------------------
+
+Once the status is ``success``, you can download the export onto your local machine. First, discover the name of the completed export file with ``mmctl export list``:
+
+.. code::
+
+   $ mmctl export list
+   r3kcj8yuwbramdt714doafi3oo_export.zip
+
+This will show all of the exports on the server, so be sure to download the latest one and to delete it when you're done to save storage. Download the file with a command like the following, but with the filename of the export on your server:
+
+.. code::
+
+   $ mmctl export download r3kcj8yuwbramdt714doafi3oo_export.zip
+
+Upload the export to the new server
+-----------------------------------
+
+Finally, it's time to take our export from the source server and use it as an import into the destination server. First, log into the destination server using mmctl the same way you logged into the source server:
+
+.. code::
+
+   $ mmctl auth login https://localinstance.company.com
+
+Use the following command to upload the export to the destination server. The speed of the upload may vary based on connection speed. When the upload is complete the command will return with the ID of the import:
+
+.. code::
+  
+   mmctl import upload r3kcj8yuwbramdt714doafi3oo_export.zip
```

Comment:
One thing to note that I discovered after writing my blog post is that this step is subject to the upload file size limit specified in the System Console. On Cloud, and by default I believe, this is 100MB. For self-hosted -> Cloud, if the import is >100MB in size, the user will get stuck and have to contact support at this step. For Cloud -> Self-Hosted, the user may hit a bump here and need to go into the System Console to raise their upload limit to something larger than the attachment upload.

I'm 0/5 on this but long term I think it would be worth investigating whether or not uploaded archives being sent via `mmctl` should be subject to the same file upload limit as files that are uploaded into Channels via the web interface.

Reply:
Great point, @gigawhitlocks! @justinegeffen @nab-77 for visibility

Reply:
Thanks for bringing this up. I did see your comment about this but it slipped my mind. 

I think one way of circumventing this is to recommend changing the limit prior to the process of cloud > self-hosted. Do we know the average size of the upload? 

Assuming it's not possible to change this setting for Cloud, what happens to the user's process? Do they need to restart?

Reply:
For imports to Cloud, they'll have to reach out to support and have the process handled by us. Support can upload the archive to the AWAT which will import it in a way that is not subject to this limitation. Their workspace won't be in a bad state or anything, it'll just be empty. Upload will fail, is all.

>one way of circumventing this is to recommend changing the limit prior to the process of cloud > self-hosted. Do we know the average size of the upload?

The only way to circumvent it is to change the limit prior to the upload. The size of the upload is entirely dependent upon the size of the source installation, so there's no way to estimate what an average size might be without collecting a lot of data from on-premise installations. After the export is generated, the user should look at the archive in a file manager (like Windows Explorer or macOS Finder) to see how big it is, and raise the limit accordingly, before attempting to upload it to the destination.